Automattic/jetpack
# [Boost] - Improve the Modal Messages

### Description
This builds on the previous work to add a score drop modal on top of the rating modal. It centralises the code so we can add further modal pop outs and only calculates the score change once (previously it calculated it twice).

I also modified the wording to make it simpler for a score increase. Previously we had different wording if it went up by a higher amount, which while cool I'm not convinced was necessary and this makes it easier to manage. 

#### Testing instructions:
Difficult to test live since you'll need scores that rises & falls but you can modify the value of `changePercentage` in `getScoreMovementPercentage` in `/app/assets/src/js/api/speed-scores.ts` to return `10` or `-10` and then see if the modal returns after it's dismissed. 

It's worth noting that the "X" still only removes the modal, where's clicking the CTA or the "do not show me this again" banishes the modal. So don't expect the "X" on the modal to banish the modal for good. 

* Modify the change percentage value to 10 and refresh the page - the score has increased modal will show
* Click the "x" button on the modal this will dismiss it.
* Refresh the page , it'll come back. This time click the "do not show me again".
* "Refresh" the speed scores (not the page). The modal will not return. 
* Refresh the page. The modal will not return. 
* Modify the change percentage value to -1. No modal will show. 
* Modify the change percentage value to -20. The score drop modal will show. 
* Repeat various dismiss & refresh combinations. It should reappear only if "x" is clicked. 

#### Does this pull request change what data or activity we track or use?
No.
